include/grpc++/impl/codegen/proto_utils.h

```diff
@@ -175,64 +175,87 @@ class GrpcBufferReader final
     return byte_count_ - backup_count_;
   }
 
- private:
+ protected:
   int64_t byte_count_;
   int64_t backup_count_;
   grpc_byte_buffer_reader reader_;
   grpc_slice slice_;
   Status status_;
 };
+
+// BufferWriter must be a subclass of io::ZeroCopyOutputStream.
```

Comment:
static_assert(std::is_base_of(BufferWriter, io::ZeroCopyOutputStream), "BufferWriter must be a subclass of io::ZeroCopyOutputStream");